File: src/code-samples/langsmith/smithdb-migration/runs-query-basic-before.ts
// :remove-start:
if (!process.env.LANGSMITH_API_KEY) {
  console.log("Skipping (LANGSMITH_API_KEY required).");
  process.exit(0);
}
process.on("uncaughtException", (reason) => {
  const msg = String(reason);
  if (
    /\b[45]\d{2}\b/.test(msg) ||
    msg.includes("Invalid UUID") ||
    msg.includes("PermissionDenied")
  ) {
    console.log(`Skipping (placeholder values in test env): ${reason}`);
    process.exit(0);
  }
  process.exit(1);
});
// :remove-end:

// :snippet-start: runs-query-basic-before-js
// :codegroup-tab: Before
import { Client } from "langsmith";

const client = new Client();
const runs = client.listRuns({
  projectName: "default",
  runType: "llm",
  filter: 'and(eq(status, "success"), gt(total_tokens, 100))',
  limit: 100,
});
for await (const run of runs) {
  console.log(run.id, run.name, run.status);
}
// :snippet-end:

File: src/code-samples/langsmith/smithdb-migration/runs-query-basic-before.kt
///usr/bin/env jbang "$0" "$@" ; exit $?
//JAVA 21
//KOTLIN 2.0.21
//DEPS com.langchain.smith:langsmith-java:0.1.0-beta.8

// :snippet-start: runs-query-basic-before-kt
// :codegroup-tab: Before
import com.langchain.smith.client.LangsmithClient
import com.langchain.smith.client.okhttp.LangsmithOkHttpClient
import com.langchain.smith.models.runs.RunQueryV1Params
import com.langchain.smith.models.runs.RunTypeEnum

// :remove-start:
fun main() {
    if (System.getenv("LANGSMITH_API_KEY").isNullOrBlank()) {
        println("[smithdb-runs-query-basic-before] Skipping (LANGSMITH_API_KEY is not set).")
        return
    }

// :remove-end:
val client: LangsmithClient = LangsmithOkHttpClient.fromEnv()

val params = RunQueryV1Params.builder()
    .addSession("<project-id>")
    .runType(RunTypeEnum.LLM)
    .filter("""and(eq(status, "success"), gt(total_tokens, 100))""")
    .addSelect(RunQueryV1Params.Select.ID)
    .addSelect(RunQueryV1Params.Select.NAME)
    .addSelect(RunQueryV1Params.Select.STATUS)
    .limit(100L)
    .build()

client.runs().queryV1(params).runs().forEach { run ->
    println("${run.id()} ${run.name()} ${run.status()}")
}
// :remove-start:
}
// :remove-end:
// :snippet-end:
